### PR TITLE
[camerax] Add fix for camera preview rotation on landscape-oriented devices and set up fix for Impeller support

### PR DESCRIPTION
**Pull request by camsim99**
Partially lands https://github.com/flutter/packages/pull/6856. This PR specifically includes:

#### 1: A fix for correctly rotating the camera preview
This fix is required for `Surface`s not backed by a `SurfaceTexture` because they don't contain the transformation information needed to correctly rotate the camera preview. In that case, we use the logic described in https://developer.android.com/media/camera/camera2/camera-preview#orientation_calculation.

The fix  is **not currently used** (the logic is not reachable) as Impeller support has not been added back to the plugin, but has been tested in https://github.com/flutter/packages/pull/6856 and will be turned on when https://github.com/flutter/flutter/issues/149294 is fixed.

Part of https://github.com/flutter/flutter/issues/149294.

#### 2: A fix for correctly rotating the camera preview on naturally landscape-oriented devices
I believe this issue was caused because we assume that the natural orientation of the device is portrait up. We fix this here by adding an extra rotation for the camera preview based on the natural orientation of the device.

Fixes https://github.com/flutter/flutter/issues/149177.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart [hunk 1]
```diff
@@ -366,6 +394,26 @@ class AndroidCameraCameraX extends CameraPlatform {
...
+          .getSensorOrientation(camera2CameraInfo)
+          .then((int value) => sensorOrientation = value),
+      proxy
+          .getUiOrientation()
+          .then((DeviceOrientation value) => naturalOrientation ??= value),
+    ]);
+    _subscriptionForDeviceOrientationChanges = onDeviceOrientationChanged()
+        .listen((DeviceOrientationChangedEvent event) {
+      currentDeviceOrientation = event.orientation;
+    });
```
Comment (camsim99) [inline packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart:415]: @matanlurey if you have a suggestion as to how to make these run in parallel, do let me know!

Comment (matanlurey) [reply packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart:415]: Ah I didn't notice you already had Future.wait, honestly this is fine. LGTM.

**Review by matanlurey — APPROVED** [review]

**Comment by camsim99** [comment]
> @camsim99 Could we have a similar fix with the `video_player_android`?

Can you file an issue for this? I'm assuming you're suggesting a fix for landscape-oriented devices because `video_player_android` received Impeller support in https://github.com/flutter/packages/pull/6989.

**Comment by AlexV525** [comment]
> Can you file an issue for this? I'm assuming you're suggesting a fix for landscape-oriented devices because `video_player_android` received Impeller support in #6989.

https://github.com/flutter/flutter/issues/154696

**Pull request merged**